Source/DataSourceProvider.swift

```diff
@@ -38,7 +38,9 @@ where CellFactory.Item == DataSource.Item, SupplementaryFactory.Item == DataSour
 
     fileprivate var bridgedDataSource: BridgedDataSource?
 
-
+    //The data source that provides the editing functionality on the tableView
+    public var dataSourceTableEditing: DataSourceTableEditingProtocol?
```

Comment:
I believe we can add that requirement here 
`public extension DataSourceProvider where CellFactory.View: UITableViewCell`.
what do you think?